### PR TITLE
feat(analytics): Add analytics events for "choose your own adventure" flow

### PR DESCRIPTION
**Pull request by jophish**
### Description

For ACT-663.

Implements all the analytics events, instrumented with the proper data, that we'll need in order to satisfy ACT-663. Metrics still need to be set up in Statsig, however.

### Test plan

Unit tested.

### Related issues

- Fixes ACT-663.

### Backwards compatibility

Yes.

### CODE REVIEW

File: src/analytics/Properties.tsx [hunk 1]
```diff
@@ -60,6 +60,8 @@ import { RecipientType } from 'src/recipients/recipient'
 import { Field } from 'src/swap/types'
 import { Currency, CurrencyOrCREAL, StableCurrency } from 'src/utils/currencies'
 import { Awaited } from 'src/utils/typescript'
+import { AdventureCardMap } from 'src/onboarding/ChooseYourAdventure'
```
Comment (satish-ravi) [inline src/analytics/Properties.tsx:63]: I guess we have other screen imports here that uses Analytics, but this introduces a cyclic dependency right? 
```
ChooseYourAdventure -> ValoraAnalytics -> Properties -> ChooseYourAdventure
```
Should we avoid introducing another one?

File: src/analytics/Properties.tsx [hunk 2]
```diff
@@ -336,6 +338,12 @@ interface OnboardingEventsProperties {
   [OnboardingEvents.protect_wallet_help_dismiss]: undefined
   [OnboardingEvents.protect_wallet_copy_phrase]: undefined
   [OnboardingEvents.protect_wallet_complete]: undefined
+  [OnboardingEvents.cya_button_press]: {
+    name: string
+    index: number
+    indexMap: AdventureCardMap
```
[the remark this answers is not in the document]
Comment (jophish) [reply src/analytics/Properties.tsx:344]: Hmm, no real benefit. I'll switch it to an array and make the index 1-based. I can rename it to something like `position` also.

File: src/analytics/Properties.tsx (hunk 1 above)
Comment (jophish) [reply src/analytics/Properties.tsx:63]: Yeah, I can refactor things in order to avoid this cycle.

**Review by jh2oman — APPROVED** [review]
Looks good! The data seem sufficient to calculate all the metrics

**Pull request merged**